src/lib/jsonParseV2/parseSubscription.cpp

```diff
@@ -30,50 +30,44 @@
 #include "alarmMgr/alarmMgr.h"
 #include "common/globals.h"
 #include "common/RenderFormat.h"
+#include "common/string.h"
 #include "rest/ConnectionInfo.h"
 #include "rest/OrionError.h"
-#include "ngsi/ParseData.h"
 #include "ngsi/Request.h"
 #include "jsonParseV2/jsonParseTypeNames.h"
 #include "jsonParseV2/jsonRequestTreat.h"
+#include "jsonParseV2/utilsParse.h"
+#include "rest/Verb.h"
+
+
 #include "jsonParseV2/parseSubscription.h"
 
 using namespace rapidjson;
 
-
+using ngsiv2::SubscriptionUpdate;
+using ngsiv2::EntID;
 
 /* Prototypes */
 static std::string parseAttributeList(ConnectionInfo* ciP, std::vector<std::string>* vec, const Value& attributes);
-static std::string parseNotification(ConnectionInfo* ciP, SubscribeContextRequest* scrP, const Value& notification);
-static std::string parseSubject(ConnectionInfo* ciP, SubscribeContextRequest* scrP, const Value& subject);
-static std::string parseEntitiesVector(ConnectionInfo* ciP, EntityIdVector* eivP, const Value& entities);
-static std::string parseNotifyConditionVector(ConnectionInfo* ciP, SubscribeContextRequest* scrP, const Value& condition);
-
+static std::string parseNotification(ConnectionInfo* ciP, SubscriptionUpdate* subsP, const Value& notification);
+static std::string parseSubject(ConnectionInfo* ciP, SubscriptionUpdate* subsP, const Value& subject);
+static std::string parseEntitiesVector(ConnectionInfo* ciP, std::vector<EntID>* eivP, const Value& entities);
+static std::string parseNotifyConditionVector(ConnectionInfo* ciP, SubscriptionUpdate* subsP, const Value& condition);
+static std::string error(ConnectionInfo* ciP, const std::string& msg);
```

Comment:
It's a internal (static) function, only visible inside this module. It would be more related to parsing than to rendering. But we can change its name whatever we think it's clearer ...